### PR TITLE
rewrite non-conan build instructions

### PR DESCRIPTION
**Pull request by Jaguar0625**
conan is preferred approach. 

for non-preferred approach provide instructions for linux / os x that are more closely aligned with what official builds do for easier maintenance.

### CODE REVIEW

File: docs/BUILD-manual.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# Building from source
+
+This guide should mostly work for Linux and OS X operating systems.
+It does not include instructions for Windows.
+For Windows, [Build with CONAN](BUILD-conan.md) is strongly encouraged.
+
+Assume an environment variable named `CAT_DEPS_DIR` is defined as the directory that should contain all catapult directories.
```
Comment (segfaultxavi) [inline docs/BUILD-manual.md:7]:
```suggestion
Assume an environment variable named `CAT_DEPS_DIR` is defined as the directory that should contain all catapult dependencies.
```

I assume you meant this?

**Review by segfaultxavi — CHANGES_REQUESTED** [review]
Found only one oversight. I haven't checked the correctness of the instructions.

**Comment by Jaguar0625** [comment]
@segfaultxavi - any more comments?

**Comment by segfaultxavi** [comment]
No comments regarding the text, but I haven't verified the instructions are correct.

**Comment by Jaguar0625** [comment]
Can you please verify? I'd like to wrap this up today.

**Comment by Jaguar0625** [comment]
Needed to merge before updating dependencies. 

@segfaultxavi - please open issue if there are issues with scripts/instructions.

**Pull request closed without merging**